network/peer.go

```diff
@@ -836,3 +883,62 @@ func (p *peer) getIP() utils.IPDesc {
 	defer p.ipLock.RUnlock()
 	return p.ip
 }
+
+// addAlias marks that we have found another
+// IP that we can connect to this peer at.
+//
+// assumes stateLock is held
+func (p *peer) addAlias(ip utils.IPDesc) {
+	p.aliasesLock.Lock()
+	defer p.aliasesLock.Unlock()
+
+	p.net.peerAliasIPs[ip.String()] = struct{}{}
+	p.aliases = append(p.aliases, alias{
+		ip:    ip,
+		added: p.net.clock.Time(),
+	})
+}
+
+// releaseNextAlias returns the next released
+// alias or nil if none was released.
+func (p *peer) releaseNextAlias(now time.Time) *alias {
```

Comment:
I created a separate function to release the next alias to avoid littering the logic that follows with granular lock/unlock code. I figured this was much easier to maintain.